src/Room/redux/sagas.ts

```diff
@@ -1,24 +1,21 @@
-import { put, takeLatest } from 'redux-saga/effects'
-import searchYoutube from 'youtube-api-v3-search'
+import { delay, put, takeLatest } from 'redux-saga/effects'
+import searchYoutube, { Options, Results } from 'youtube-api-v3-search'
 import { ActionCreators, types } from './types'
 import actions from './actions'
+import { youtubeDeserializer } from './deserializers'
 
 function* querySongs(
   action: ReturnType<ActionCreators['ChangeQuery']>,
 ) {
-  const options = {
+  yield delay(250)
```

Comment:
This is interesting. I assume this is to prevent the saga from firing on every keystroke. If so, what are the benefits of doing it this way vs. within the component itself `onChange` or `onKeyup` like I'd do back in the good old jQuery days?

Reply:
Great question.  Yup, this delays the execution of the saga for 250ms; we use it in conjunction with `takeLatest` (where we bind the action creator to this saga function) because `takeLatest` will cancel any previous-iterations of the same saga whenever it gets a new action (read: when someone types in a new letter in the input field and triggers this saga to run, it will cancel the previous saga that may be waiting here in the `delay`).

We could handle the delay in `onChange` or `onKeyup` instead, but we'd end up recreating the same (asynchronous) functionality with `setTimeout` and `clearTimeout`.  We've chosen the `redux-saga` as the place to handle async stuff, so it's a bit clearer if it's all concentrated in this one place.